torch/csrc/autograd/saved_variable.cpp

```diff
@@ -43,6 +43,18 @@ SavedVariable::SavedVariable(const Variable& variable, bool is_output, bool is_i
     saved_version_ = version_counter.current_version();
     is_leaf_ = variable.is_leaf();
     is_output_ = is_output;
+    is_inplace_on_view_ = is_inplace_on_view;
+
+    if(is_inplace_on_view) {
+      TORCH_INTERNAL_ASSERT(!is_leaf_ && is_output);
+      weak_grad_fn_ = variable.grad_fn();
+    }
+
+    if (get_default_hooks()) {
```

Comment:
The way this is written right now is not performant. `get_default_hooks()` constructs a unique_ptr, then throws it away, and inside the body we call `get_default_hooks()` again to construct a second unique_ptr.

We should do something like:
```
auto maybe_hooks = get_default_hooks()
if (maybe_hooks) {
  ...
  register_hooks_(std::move(maybe_hooks), variable)
}
```
but really I think we shouldn't be using unique_ptr here (see my later comment)

Reply:
I will update the PR with this, thanks! And yes I will create open a new PR to think about how to use shared_ptr instead.